pywbem/mof_compiler.py

```diff
@@ -2430,13 +2439,16 @@ def compile_file(self, filename, ns):
 
           : Any exceptions that are raised by the repository connection class.
         """
-
         if self.parser.verbose:
             self.parser.log('Compiling file ' + filename)
 
-        f = open(filename, 'r')
-        mof = f.read()
-        f.close()
+        if not os.path.exists(filename):
+            filename = os.path.basename(filename)
+            filename = self.find_mof(filename[:-4].lower())
+            if filename is None:
+                raise ValueError('No such file: %s' % filename)
```

Comment:
Before this change, the method raised IOError if the file could not be found. It now raises ValueError. This is an incompatible change for users of the MOF compiler API (e.g. our wbemcli Python script (which admittedly has an issue in handling any of these two exceptions). We should go back to raising IOError, IMO.